From 3b8b82a05987053ac5dd777ef04e054124ef8e2b Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Wed, 13 Apr 2016 19:05:20 -0500
Subject: [PATCH 06/23] dtb: imx6sxscm: add support for M4 and Linux dual boot

A new dts file imx6sxscm-1gb-evb-ldo-m4.dts is created for
supporting Linux and M4 dual booting.

To run a sample on the M4 you need to use the mmc dev 1
for the SD3 and use the proper dtb.

An example of how to change bootargs is as follows:

setenv fdt_file imx6sxscm-1gb-evb-ldo-m4.dtb

setenv mmcdev 1

setenv loadM4 'fatload mmc ${mmcdev} 0x80000000 \
               rpmsg_pingpong_freertos_example.bin; \
               dcache flush; cp.b 0x80000000 0x7F8000 0x8000; \
               dcache flush; bootaux 0x7F8000'

setenv mmcargs 'setenv bootargs console=${console},${baudrate} \
                uart_from_osc root=${mmcroot}'

setenv bootcmd 'run loadM4; mmc dev ${mmcdev};mmc dev ${mmcdev}; \
                if mmc rescan; then if run loadbootscript; \
                then run bootscript; else if run loadimage; \
                then run mmcboot; else run netboot; \
                fi; fi; else run netboot; fi'

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
---
 arch/arm/boot/dts/Makefile                     |  1 +
 arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-m4.dts | 98 ++++++++++++++++++++++++++
 2 files changed, 99 insertions(+)
 create mode 100644 arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-m4.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index dbde3af..8144ef6 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -244,6 +244,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sx-sdb-m4.dtb \
 	imx6sxscm-1gb-evb-ldo.dtb \
 	imx6sxscm-1gb-evb-ldo-lcdif1.dtb \
+	imx6sxscm-1gb-evb-ldo-m4.dtb \
 	vf610-twr.dtb \
 	imx7d-12x12-lpddr3-arm2.dtb \
 	imx7d-12x12-ddr3-arm2.dtb \
diff --git a/arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-m4.dts b/arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-m4.dts
new file mode 100644
index 0000000..eadeca9
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sxscm-1gb-evb-ldo-m4.dts
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sdb-ldo.dts"
+#include "imx6sxscm-evb.dtsi"
+
+/{
+	memory {
+		linux,usable-memory = <0x80000000 0x1ff00000>,
+				<0xa0000000 0x1ff00000>;
+	};
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+};
+
+/*
+ * The flollowing modules are conflicting with M4, disable them when m4
+ * is running.
+ */
+&adc1 {
+        status = "disabled";
+};
+
+&adc2 {
+        status = "disabled";
+};
+
+&flexcan1 {
+	status = "disabled";
+};
+
+&flexcan2 {
+	status = "disabled";
+};
+
+&i2c3 {
+        status = "disabled";
+};
+
+&mcctest{
+	status = "okay";
+};
+
+&mcctty{
+	status = "okay";
+};
+
+&ocram {
+	reg = <0x00901000 0xf000>;
+};
+
+&qspi2 {
+	status = "disabled";
+};
+
+&qspi_m4 {
+	status = "okay";
+};
+
+&rpmsg{
+	status = "okay";
+};
+
+&uart2 {
+	status = "disabled";
+};
+
+&clks {
+	fsl,shared-clks-number = <0x23>;
+	fsl,shared-clks-index = <IMX6SX_CLK_PLL2_BUS IMX6SX_CLK_PLL2_PFD0
+			IMX6SX_CLK_PLL2_PFD2 IMX6SX_CLK_PLL3_USB_OTG
+			IMX6SX_CLK_PLL3_PFD1 IMX6SX_CLK_PLL3_PFD2
+			IMX6SX_CLK_PLL3_PFD3 IMX6SX_CLK_PLL4_AUDIO
+			IMX6SX_CLK_PLL5_VIDEO
+			IMX6SX_CLK_OCRAM IMX6SX_CLK_CAN1_SERIAL
+			IMX6SX_CLK_CAN1_IPG IMX6SX_CLK_CAN2_SERIAL
+			IMX6SX_CLK_CAN2_IPG IMX6SX_CLK_CANFD
+			IMX6SX_CLK_ECSPI1 IMX6SX_CLK_ECSPI2
+			IMX6SX_CLK_ECSPI3 IMX6SX_CLK_ECSPI4
+			IMX6SX_CLK_ECSPI5 IMX6SX_CLK_QSPI1
+			IMX6SX_CLK_QSPI2 IMX6SX_CLK_SSI1
+			IMX6SX_CLK_SSI2 IMX6SX_CLK_SSI3
+			IMX6SX_CLK_UART_SERIAL IMX6SX_CLK_UART_IPG
+			IMX6SX_CLK_PERIPH_CLK2_SEL IMX6SX_CLK_OCRAM_ALT_SEL
+			IMX6SX_CLK_I2C1 IMX6SX_CLK_I2C2
+			IMX6SX_CLK_I2C3 IMX6SX_CLK_I2C4
+			IMX6SX_CLK_EPIT1 IMX6SX_CLK_EPIT2>;
+	fsl,shared-mem-addr = <0x91F000>;
+	fsl,shared-mem-size = <0x1000>;
+};
-- 
1.9.1

